### ISSUE TITLE
Deferred props can't be SSR'd

### ISSUE THREAD

**Issue by JenteSL** [inertiajs/inertia#2150]
### Versions:

- `@inertiajs/core` version: 2.0.0
- `@inertiajs/react` version: 2.0.0

----

### Describe the problem:

When enabling server-side rendering (SSR) and disabling JavaScript in the browser, `<Deferred />` components are displayed using their fallback component instead of the resolved children.

I believe this is because the Router's [`loadDeferredProps`](https://github.com/inertiajs/inertia/blob/19b6ab40eca82d900c2f0b94e62fb4f1847c073d/packages/core/src/router.ts#L358) relies on [`reload`](https://github.com/inertiajs/inertia/blob/19b6ab40eca82d900c2f0b94e62fb4f1847c073d/packages/core/src/router.ts#L88), which is a browser-only method.

Is there a workaround for resolving the `<Deferred />` components when server-side rendering?

------

### Steps to reproduce:

In a new Laravel example with Inertia + React installed (using default `app.jsx` and `ssr.jsx`):

```php
# routes/web.php

Route::get('/message', function () {
    return Inertia::render('Message', [
        'message' => Inertia::defer(fn () => 'Hello world'),
    ]);
});
```

```jsx
// resources/js/Pages/Message.jsx

import { Deferred } from '@inertiajs/react'

export default function Message({ message }) {
    return (
        <Deferred data="message" fallback={<p>Loading...</p>}>
            <DeferredMessage />
        </Deferred>
    );
}

import { usePage } from '@inertiajs/react'

function DeferredMessage() {
    const message = usePage().props.message;

    return (
        <p>
            {message}
        </p>
    );
}

```

Then when doing a vite build (a regular one and an `--ssr` one, using `npm run build`) and starting the ssr-server (using `php artisan inertia:start-ssr`), going to '/message` will show up as:

```
Loading...
```

instead of:

```
Hello world
```

**Comment by RobertBoes** [comment]
Wouldn't this be desired behavior (intended or not)? Deferred props are used to defer loading of props, mostly for performance benefits. If during SSR they would be loaded then the initial response would take longer, defeating the purpose of using deferred props

**Comment by JenteSL** [comment]
Thanks for the answer.

So I learned something—it makes sense for the Deferred components to show the fallback as a server-rendered starting point that then gets hydrated + completed (fetching the deferred props).

Coming from Remix, I thought the behavior there was different as to allow search engines to index the full page (including resolved deferred components).

But it turns out that Remix also shows the fallback when disabling JavaScript in the browser. To allow search engines to index the full page, they instead rely on [checking whether the request comes from a bot](https://github.com/remix-run/remix/blob/542310d95cba4ac41eb70cbd26a7c40a9e31090e/templates/remix/app/entry.server.tsx#L27) and, if so, return the full HTML document.

For this they use [a React streaming API that waits for all content to be loaded](https://react.dev/reference/react-dom/server/renderToPipeableStream#waiting-for-all-content-to-load-for-crawlers-and-static-generation). 

But since Inertia’s deferred properties are not actually streaming, but partial reloads, maybe there isn't an easy solution to wait for all content to be loaded in the case of a bot?

**Comment by pascalbaljet** [comment]
Yes, this is the desired behavior for SSR as explained by Robert :)
